### PR TITLE
Initial snapshots docs

### PR DESCRIPTION
**Pull request by simon-bar**
Can be merged already, if there are no more remarks referring the last section about uploading the snapshot.

### CODE REVIEW

File: source/includes/_snapshots.md [hunk 1]
```diff
@@ -1 +1,126 @@
 # Snapshots
+
+## The Snapshot Resource
+
+### Attributes:
+
+
+Attribute | Type | Description
+--------- | ------- | -------
+`id` | string | A unique identifier of this snapshot.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: `the` snapshot

File: source/includes/_snapshots.md [hunk 2]
```diff
@@ -1 +1,126 @@
...
+`created_at` | datetime | The time the snapshot was uploaded to or created on the manager.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: Please change datetime to ISO 8601 as it appears here: http://docs.getcloudify.org/api/#events

File: source/includes/_snapshots.md [hunk 3]
```diff
@@ -1 +1,126 @@
...
+`status` | string | Status of this snapshot - created/failed/uploading/uploaded.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: Status of the snapshot. One of created/failed/uploading/uploaded.

File: source/includes/_snapshots.md [hunk 4]
```diff
@@ -1 +1,126 @@
...
+            "size": 10000,
+            "total": 2
+        }
+    }
+}
+```
+
+### Response
+
+Field | Type | Description
```
Comment (idanmo) [inline source/includes/_snapshots.md]: `Attribute` instead of Field

File: source/includes/_snapshots.md [hunk 5]
```diff
@@ -1 +1,126 @@
...
+--------- | ------- | -------
+`items` | list | A list of `Snapshot` resources.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: You can have a link to the `Snapshot` resource.. use [Snapshot](#the-snapshot-resource)

File: source/includes/_snapshots.md [hunk 6]
```diff
@@ -1 +1,126 @@
...
+`metadata` | object | A dictionary containing data useful for pagination logic.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: You don't have to include metadata here as we have it documented separately.

File: source/includes/_snapshots.md [hunk 7]
```diff
@@ -1 +1,126 @@
...
+
+
+## Create Snapshot
+`PUT /api/v2/snapshots/{snapshot-id}`
+
+Creates a new snapshot.
+
+### URI Parameters
+* snapshot-id: The id of the new snapshot
```
Comment (idanmo) [inline source/includes/_snapshots.md]: "." at the end of the sentence.

File: source/includes/_snapshots.md [hunk 8]
```diff
@@ -1 +1,126 @@
...
+
+### Request Body
+Property | Default | Description
+--------- | ------- | -----------
+`include_metrics` | False | Should InfluxDB data be included in the snapshot.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: `false` (lowercase)

"Specifies whether metrics stored in InfluxDB should be included in created snapshot."

File: source/includes/_snapshots.md [hunk 9]
```diff
@@ -1 +1,126 @@
...
+`include_credentials` | False | Should agent keys, even those specified in uploaded blueprints, be included in the snapshot.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: `false` (lowercase)

"Specifies whether agent SSH keys (including those specified in uploaded blueprints) should be included in the created snapshot.

File: source/includes/_snapshots.md [hunk 10]
```diff
@@ -1 +1,126 @@
...
+
+### Response
+An `Execution` resource representing the create snapshot workflow execution.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: Link to Execution resource (as mentioned before).

File: source/includes/_snapshots.md [hunk 11]
```diff
@@ -1 +1,126 @@
...
+
+
+## Delete Snapshot
+`DELETE /api/v2/snapshots/{snapshot-id}`
+
+Deletes an existing snapshot
```
Comment (idanmo) [inline source/includes/_snapshots.md]: "." at the end.

File: source/includes/_snapshots.md [hunk 12]
```diff
@@ -1 +1,126 @@
...
+
+### URI Parameters
+* snapshot-id: The id of the snapshot to be deleted
```
Comment (idanmo) [inline source/includes/_snapshots.md]: "." at the end.

File: source/includes/_snapshots.md [hunk 13]
```diff
@@ -1 +1,126 @@
...
+
+### Response
+An empty `Snapshot` resource, with one non-empty field (its id).
```
Comment (idanmo) [inline source/includes/_snapshots.md]: Link to snapshot resource.

File: source/includes/_snapshots.md [hunk 14]
```diff
@@ -1 +1,126 @@
...
+
+## Restore Snapshot
+`POST /api/v2/snapshots/{snapshot-id}/restore`
+
+### URI Parameters
+* snapshot-id: The id of the snapshot to be restored
```
Comment (idanmo) [inline source/includes/_snapshots.md]: "."

File: source/includes/_snapshots.md [hunk 15]
```diff
@@ -1 +1,126 @@
...
+
+### Request Body
+Property | Default | Description
+---------|---------|-------------
+`force`  |  False  | Force restoring the snapshot on a manager that already contains blueprints and/or deployments
```
Comment (idanmo) [inline source/includes/_snapshots.md]: `false`

"Specifies whether to force restoring the snapshot on a manager that already contains blueprints/deployments".

File: source/includes/_snapshots.md [hunk 16]
```diff
@@ -1 +1,126 @@
...
+`recreate_deployments_envs` | True | Should deploments environments for restored deployments be created - this is rather natural and strongly recommended.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: `true`

"Specifies whether deployment environments should be created for restored deployments."

File: source/includes/_snapshots.md [hunk 17]
```diff
@@ -1 +1,126 @@
...
+
+### Response
+An `Execution` resource representing the restore snapshot workflow execution.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: Link to execution resource.

File: source/includes/_snapshots.md [hunk 18]
```diff
@@ -1 +1,126 @@
...
+
+
+## Download Snapshot
+`GET /api/v2/snapshots/{snapshot-id}/archive`
+
+Downloads an existing snapshot.
+
+### URI Parameters
+* snapshot-id: The id of the snapshot to be restored
```
Comment (idanmo) [inline source/includes/_snapshots.md]: "."

File: source/includes/_snapshots.md [hunk 19]
```diff
@@ -1 +1,126 @@
...
+
+### Response
+A streamed response (content type `application/octet-stream`), which is a zip archive.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: ... containing the snapshot data.

File: source/includes/_snapshots.md [hunk 20]
```diff
@@ -1 +1,126 @@
...
+
+## Upload Snapshot
+`PUT /api/v2/snapshots/{snapshot-id}/archive`
+
+Uploads a snapshot to the Cloudify Manager.
+The call expects a "application/octet-stream" content type where the content is a zip archive.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: highlight `application/octet-stream`

File: source/includes/_snapshots.md [hunk 21]
```diff
@@ -1 +1,135 @@
...
+The call expects a `application/octet-stream` content type where the content is a zip archive.
+It is possible to upload a snapshot from a URL by specifying the URL in the `snapshot_archive_url` request body property.
+
+### URI Parameters
+* snapshot-id: The id of the snapshot to be uploaded.
+
+### Request Body
+Property | Default | Description
+---------|---------|-------------
+`snapshot_archive_url`  |   | Optional parameter specifying a url to a snapshot that will be uploaded to the manager.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: AFAIK in request body we specify the type of the parameter and not the default value.
Look at blueprints upload for example...

Comment (simon-bar) [inline source/includes/_snapshots.md]: Hmm so this comment applies for the previous tables where we were correcting False to false and so on, right?

File: source/includes/_snapshots.md [hunk 22]
```diff
@@ -1 +1,135 @@
...
+### URI Parameters
+* snapshot-id: The id of the snapshot to be uploaded.
+
+### Request Body
+Property | Type | Description
+---------|---------|-------------
+`snapshot_archive_url` | string | Optional parameter specifying a url to a snapshot that will be uploaded to the manager.
+
+### Response
+A [`Snapshot`](#the-snapshot-resource) resource.
```
Comment (idanmo) [inline source/includes/_snapshots.md]: The back-ticks here are not needed as this is a link.
Please fix in all places where you have a link.

**Pull request merged**